pnpm-lock.yaml

```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.0'
+lockfileVersion: '9.0'
```

Comment:
pnpm 6 says it's not compatible with Node 21 used in the CI workflow, so I used the latest.
That however changed the format of the lock file slightly

Reply:
I see that CI actually installs v9

<img width="648" alt="Screenshot 2024-08-07 at 20 03 39" src="https://github.com/user-attachments/assets/596bbb8b-0bba-4bd4-8717-d4b396bad305">

but v9 changes the format and the version of the lock file when dependencies changed.
kinda side effect.

Reply:
found the explanation:
https://github.com/pnpm/pnpm/issues/7934#issuecomment-2059106340

Reply:
looks like nothing I can do about it to minimize the diff of the lock

Reply:
pnpm v6 != pnpm lockfile version 6.
for example pnpm v8 uses lockfile version 6.
The package.json lists pnpm requirement as >= v8, but 8 does not have support for lockfile version 9. (I don't know where a good reference is for pnpm lockfile versions.)
I thought pnpm v8 would work with Node 21; so, was there confusion about lockfile version versus runtime version and lockfile should go back down to allow working with pnpm v8 OR should the package.json be updated to require pnpm v9?
cc: @andrewbranch